### ISSUE TITLE
Wrong PositionalEncoding in the Transformer example

### ISSUE THREAD

**Issue by Galaxy-Husky** [Lightning-AI/pytorch-lightning#19138]
### Bug description

Hi,

I think there are several mistakes in the implemention of the PositionalEncoding in the [lightning/pytorch/demos/transformer.py.](https://github.com/Lightning-AI/pytorch-lightning/blob/master/examples/fabric/language_model/train.py)

Since the transformer is set to
https://github.com/Lightning-AI/pytorch-lightning/blob/275822d5aff848b7f0a4298498e905457b8455c9/src/lightning/pytorch/demos/transformer.py#L47
1. The shape of the pe should be [batch_size, len, dim]. This transpose operation is redundant.
https://github.com/Lightning-AI/pytorch-lightning/blob/275822d5aff848b7f0a4298498e905457b8455c9/src/lightning/pytorch/demos/transformer.py#L97
2. And the shape of `self.pe` should be `self.pe[:, :x.size(1)]`
https://github.com/Lightning-AI/pytorch-lightning/blob/275822d5aff848b7f0a4298498e905457b8455c9/src/lightning/pytorch/demos/transformer.py#L88
4. The result of the addition above is not assigned to x, so the pe will not take effect.  

Would you please let me know if I'm wrong?

### What version are you seeing the problem on?

master

### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>

cc @borda

**Comment by NicholasKiefer** [comment]
You are not on the master branch.

**Comment by awaelchli** [comment]
@Galaxy-Husky Do you want to make the adjustments in a PR?

**Comment by awaelchli** [comment]
I think it would be good to go over the example one more time to fix the correctness issues. The initial version was just ported from the PyTorch examples repo but the goal then was not to make it train well but to serve as dummy example for quick testing.

**Comment by Galaxy-Husky** [comment]
> @Galaxy-Husky Do you want to make the adjustments in a PR?

Sure, I'd love to.

**Comment by Galaxy-Husky** [comment]
> I think it would be good to go over the example one more time to fix the correctness issues. The initial version was just ported from the PyTorch examples repo but the goal then was not to make it train well but to serve as dummy example for quick testing.

Thank you for your explanation.  I got it !

**Comment by manu-chauhan** [comment]
Help needed for this ? I see this is still open.
